### ISSUE TITLE
`<atomic>`: Building a header unit with `/ZI` emits warning C5260

### ISSUE THREAD

**Issue by orikama** [microsoft/STL#3287]
```
C:\Test>cl /std:c++20 /EHsc /W4 /WX /ZI /Od /exportHeader /headerName:angle atomic
Microsoft (R) C/C++ Optimizing Compiler Version 19.35.32124 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

atomic
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.35.32124\include\atomic(325): error C2220: the following warning is treated as an error
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.35.32124\include\atomic(325): warning C5260: the constant variable 'std::memory_order const (* `std::memory_order __cdecl std::_Combine_cas_memory_orders(std::memory_order,std::memory_order)'::`2'::_Combined_memory_orders)[6]' has internal linkage in an included header file context, but external linkage in imported header unit context; consider declaring it 'inline' as well if it will be shared across translation units, or 'static' to express intent to use it local to this translation unit
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.35.32124\include\atomic(325): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
```
Noticed this warning while reporting some other bugs in VS 2022 17.5.0 Preview 2

**Comment by CaseyCarter** [comment]
Line 325 defines a function-local `static constexpr` array of `memory_order` in an `inline` function:

https://github.com/microsoft/STL/blob/4483e87f6e5aa5c4598404fc01eecb7e547dd439/stl/inc/atomic#L312-L342

the name of this variable should have _no_ linkage. I smell a compiler bug. @cdacamar, do you agree?

**Comment by cdacamar** [comment]
I concur, the compiler should not be complaining here.  Put it on my tab 😉.

**Comment by StephanTLavavej** [comment]
Thanks, I've updated the title to note (1) this occurs when building the header unit (not importing it) and (2) this is specific to `/ZI` (debug info for edit and continue). I've filed VSO-1706387 internally for @cdacamar to keep track of, and I'll record this in the tracking issue #1694.

**Comment by StephanTLavavej** [comment]
Fixed by internal MSVC-PR-442426 for VS 2022 17.6 Preview 1. Thanks again for the report!

**Comment by luusl** [comment]
Is there any chance to merge the fix to 17.5? We have "Treat warnings as errors" active on all of our projects, which is causing the projects to stop compiling.

**Comment by StephanTLavavej** [comment]
@luusl As this is only a warning, I doubt it would meet the servicing bar. (There's a pretty high requirement to backport compiler changes, since every backport risks destabilizing something else.) I recommend disabling this warning as a temporary workaround until 17.6 ships; it's a very new warning so you aren't giving up any other coverage.
